src/backend/distributed/metadata/metadata_utility.c

```diff
@@ -223,19 +274,37 @@ DistributedTableSizeOnWorker(WorkerNode *workerNode, Oid relationId, char *sizeQ
 
 	if (queryResult != 0)
 	{
-		ereport(ERROR, (errcode(ERRCODE_CONNECTION_FAILURE),
-						errmsg("cannot get the size because of a connection error")));
+		ereport(logLevel, (errcode(ERRCODE_CONNECTION_FAILURE),
+						   errmsg("cannot get size of table \"%s\" from %s:%d",
+								  get_rel_name(relationId), workerNodeName,
+								  workerNodePort)));
+
+		return false;
 	}
 
 	List *sizeList = ReadFirstColumnAsText(result);
+	if (list_length(sizeList) != 1)
+	{
+		PQclear(result);
+		ClearResults(connection, failOnError);
+
+		ereport(logLevel, (errcode(ERRCODE_CONNECTION_FAILURE),
+						   errmsg("cannot get size of table \"%s\" from %s:%d",
```

Comment:
should this error be different than the above one because here it is slightly different than a connection failure I guess, which is captured above.